pkg/envoy/eds.go

```diff
@@ -0,0 +1,97 @@
+package envoy
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+
+	envoy_config_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/config/endpoint/v3"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*HasDestinationEndpointsCheck)(nil)
+
+// HasDestinationEndpointsCheck implements common.Runnable
+type HasDestinationEndpointsCheck struct {
+	*v1.Pod
+	ConfigGetter
+}
+
+// Run implements common.Runnable
+func (l HasDestinationEndpointsCheck) Run() error {
+	if l.ConfigGetter == nil {
+		log.Error().Msg("Incorrectly initialized ConfigGetter")
+		return ErrIncorrectlyInitializedConfigGetter
+	}
+	envoyConfig, err := l.ConfigGetter.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	if envoyConfig == nil {
+		return ErrEnvoyConfigEmpty
+	}
```

Comment:
Not necessary if L37-L39 exists

Reply:
This block could indeed be removed!
Thank you @shalier !!!